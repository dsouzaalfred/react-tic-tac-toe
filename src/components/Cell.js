import React from 'react';

import cn from 'classnames';
import PropTypes from 'prop-types';

import styles from './cell.module.css';

export const Cell = ({id, value, isWinningCell}) => {
  return (
    <div id={id} className={cn(styles.cell, {[styles.winner]: isWinningCell})}>
      {value === null ? '' : value}
    </div>
  );
};

Cell.propTypes = {
  id: PropTypes.string,
  value: PropTypes.string,
  isWinningCell: PropTypes.bool,
};
