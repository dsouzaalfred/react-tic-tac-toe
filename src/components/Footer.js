import React from 'react';
import styles from './footer.module.css';

export const Footer = () => {
  return (
    <div className={styles.footer}>
      By{' '}
      <a
        className={styles.link}
        href="https://dsouzaalf.red/"
        target="_blank"
        rel="noopener noreferrer"
      >
        Alfred DSouza
      </a>
    </div>
  );
};
