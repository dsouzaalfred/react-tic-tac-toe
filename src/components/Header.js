import styles from "./header.module.css";

export const Header = ({isWon, isDraw, nowPlaying, reset}) => {
  return (
    <div className={styles.header}>
      {!isWon && !isDraw && <p>Now playing: <span className={styles.note}>{nowPlaying}</span></p>}
      {isWon && <p>Winner: <span className={styles.note}>{nowPlaying}</span></p>}
      {isDraw && <p>It's a draw</p>}
      <button className={styles.button} onClick={reset}>Reset</button>
    </div>
  );
};
