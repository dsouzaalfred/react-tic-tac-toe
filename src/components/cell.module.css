@import '../vars.css';

.cell {
  height: 100px;
  width: 100px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 45px;
  text-transform: uppercase;
}
.cell:nth-child(2) {
  border-left: 1px solid var(--color-black);
}
.cell:last-child {
  border-left: 1px solid var(--color-black);
}
.cell:hover {
  background-color: var(--cell-hover);
}

.winner, .winner:hover {
  background-color: var(--winner-cell-background);
  color: var(--winner-cell-font-colour);
}
