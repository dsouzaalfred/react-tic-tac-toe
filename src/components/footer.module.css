@import '../vars.css';

.footer {
  padding: 5px;
  margin: 10px 0 0 0;
  background: var(--header-footer-background);
  width: calc(100% - 10px);
}
.link {
  color: var(--link-colour);
}
