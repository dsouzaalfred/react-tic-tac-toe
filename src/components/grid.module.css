@import '../vars.css';

.row {
  display: flex;
  border-bottom: 1px solid var(--color-black);
  max-width: 302px;
}
.row:last-child {
  border-bottom: none;
}
