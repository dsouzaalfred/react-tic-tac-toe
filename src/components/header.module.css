@import '../vars.css';

.header {
  display: flex;
  flex-flow: row;
  background-color: var(--header-footer-background);
  align-items: center;
  padding: 5px;
  justify-content: space-between;
  margin: 0 0 10px 0;
  width: calc(100% - 10px);
}

.header p {
  font-size: 1rem;
  color: var(--color-black);
}

.button {
  padding: 10px;
  width: 100px;
  background-color: var(--link-colour);
  border: 1px solid var(--link-colour);
  font-size: 1rem;
}

.note {
  color: var(--note-colour);
  font-weight: bold;
}
