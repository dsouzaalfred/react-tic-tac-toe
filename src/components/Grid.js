import { Cell } from "./Cell";

import styles from "./grid.module.css";

export const Grid = ({ handleClick, gameState, winningCells, isWon }) => {
  return (
    <div onClick={handleClick}>
      <div className={styles.row}>
        <Cell id="c1" value={gameState.c1} isWinningCell={ isWon && winningCells.includes('c1')} />
        <Cell id="c2" value={gameState.c2} isWinningCell={ isWon && winningCells.includes('c2')} />
        <Cell id="c3" value={gameState.c3} isWinningCell={ isWon && winningCells.includes('c3')} />
      </div>
      <div className={styles.row}>
        <Cell id="c4" value={gameState.c4} isWinningCell={ isWon && winningCells.includes('c4')} />
        <Cell id="c5" value={gameState.c5} isWinningCell={ isWon && winningCells.includes('c5')} />
        <Cell id="c6" value={gameState.c6} isWinningCell={ isWon && winningCells.includes('c6')} />
      </div>
      <div className={styles.row}>
        <Cell id="c7" value={gameState.c7} isWinningCell={ isWon && winningCells.includes('c7')} />
        <Cell id="c8" value={gameState.c8} isWinningCell={ isWon && winningCells.includes('c8')} />
        <Cell id="c9" value={gameState.c9} isWinningCell={ isWon && winningCells.includes('c9')} />
      </div>
    </div>
  );
};
