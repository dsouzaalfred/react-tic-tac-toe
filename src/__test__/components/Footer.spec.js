import React from 'react'
import renderer from 'react-test-renderer';

import { Footer } from '../../components/Footer';

it('Renders Cell correctly', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
