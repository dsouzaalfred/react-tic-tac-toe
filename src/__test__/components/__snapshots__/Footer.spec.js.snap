// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Cell correctly 1`] = `
<div
  className="footer"
>
  By
   
  <a
    className="link"
    href="https://dsouzaalf.red/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Alfred DSouza
  </a>
</div>
`;
